base/applications/cmdutils/attrib/attrib.c

```diff
@@ -180,75 +177,96 @@ ChangeAttribute(
     DWORD  dwAttribute;
     WCHAR  szFullName[MAX_PATH];
     LPWSTR pszFileName;
+    BOOL bWildcard = (wcschr(pszFile, L'*') || wcschr(pszFile, L'?'));
 
     /* prepare full file name buffer */
     wcscpy(szFullName, pszPath);
     pszFileName = szFullName + wcslen(szFullName);
 
-    /* change all subdirectories */
-    if (bRecurse)
-    {
-        /* append file name */
-        wcscpy(pszFileName, L"*.*");
+    /* append file name */
+    wcscpy(pszFileName, pszFile);
 
-        hFind = FindFirstFileW(szFullName, &findData);
-        if (hFind == INVALID_HANDLE_VALUE)
+    hFind = FindFirstFileW(szFullName, &findData);
+    if (hFind == INVALID_HANDLE_VALUE)
+        return FALSE;
+
+    dwAttribute = findData.dwFileAttributes;
+
+    if (!bWildcard)
+    {
+        FindClose(hFind);
+        if (dwAttribute & FILE_ATTRIBUTE_DIRECTORY)
         {
-            ErrorMessage(GetLastError(), pszFile);
-            return 1;
+            dwAttribute = (dwAttribute & ~dwMask) | dwAttrib;
+            SetFileAttributes(szFullName, dwAttribute);
+            if (!bRecurse && !bDirectories)
+            {
+                ChangeAttribute(szFullName, L"*", dwMask, dwAttrib,
+                                bRecurse, FALSE);
+            }
+            if (bRecurse && !bDirectories)
```

Comment:
These could be `else if`s since the three conditions are mutually exclusive

Reply:
OK.